launcher.py

```diff
@@ -0,0 +1,335 @@
+    parser.add_argument("--nodes", type=int, help="Number of nodes to use for the job")
+    args = parser.parse_args()
+
+    supported_base_configs = {
+        "smollm-1700M-8nodes": "examples/smollm/configs/yaml/smollm-1700M-8nodes.yaml",
+        "smollm-360M-4nodes": "examples/smollm/configs/yaml/smollm-360M-4nodes.yaml",
+        "smollm-135M-4nodes": "examples/smollm/configs/yaml/smollm-135M-4nodes.yaml",
+        "smollm-135M-1gpu": "examples/smollm/configs/yaml/smollm-135M-1gpu.yaml",
+    } # add your base configs here {name: path}
+
+    if args.base_config is None and args.config_path is None:
+        raise ValueError("Please provide a base config or a config path")
+
+
+    if args.config_path is not None and args.base_config is not None:
+        print("Both config_path and base_config are provided. Using config_path and ignoring base_config.")
+        args.base_config = None
+
+    if args.base_config not in supported_base_configs.keys():
+        raise ValueError(f"Base config {args.base_config} is not supported. Please choose one of the following: {supported_base_configs}")
+    else:
+        args.config_path = supported_base_configs[args.base_config]
+
+    if args.slurm:
+        if args.nodes is None:
+            raise ValueError("When using Slurm (--slurm), you must specify the number of nodes (--nodes)")
+
+    # Load the configuration using get_config_from_file
+    config = get_config_from_file(args.config_path, config_class=Config)
+
+    if config.general.logs_path is None and args.logs_path is None:
+        raise ValueError("Please provide a logs path")
+
+    num_params = human_format(
+        config.model.model_config.get_llama_param_count()
+    ).replace(".", ",")
+
+    if args.override:
+        for item in args.override:
+            if '=' not in item:
+                raise ValueError(f"Invalid override format: {item}. Use KEY=VALUE.")
+            key, value = item.split('=', 1)
+            try:
+                value = eval(value)
+            except:
+                pass
+            
+            set_nested_attribute(config, key, value)
+
+        print("⇄ Applied overrides:")
+        for item in args.override:
+            print(f"  {item}")
+        
+    # Calculate and print learning rate and global batch size information
+    lr_initial = config.optimizer.learning_rate_scheduler.learning_rate
+    lr_min = config.optimizer.learning_rate_scheduler.min_decay_lr
+    lr_warmup_steps = config.optimizer.learning_rate_scheduler.lr_warmup_steps
+    lr_decay_steps = config.optimizer.learning_rate_scheduler.lr_decay_steps
+    lr_decay_start = config.optimizer.learning_rate_scheduler.lr_decay_starting_step
+    lr_decay_style = config.optimizer.learning_rate_scheduler.lr_decay_style
+
+    BS = config.tokens.micro_batch_size*config.tokens.batch_accumulation_per_replica*config.tokens.sequence_length
+    GBS = BS * config.parallelism.dp
```

Comment:
To avoid confusion, maybe we can differentiate GBS and GBS * Seqlen  (call it effective_gbs or another name)?

Seems like for most people, (cf https://docs.nvidia.com/nemo-framework/user-guide/latest/nemotoolkit/nlp/nemo_megatron/batching.html and https://github.com/mosaicml/llm-foundry/blob/main/scripts/train/benchmarking/README.md#h100-80gb-bf16-large-scale--128-gpus)
gbs refers to `mbs x grad_acc x dp` only (and not with seq_len)

Reply:
changed the name + add tokens in the print to make it clear (don't want to add the one in sample not to overload it but we can add it if you think it's better)
```
📙 Training Configuration:
┌───────────────────────┬────────────────────────┐
│ Total Tokens          │                   105M │
│ Batch Size (per GPU)  │          16,384 Tokens │
│ Global Batch Size     │       1,048,576 Tokens │
└───────────────────────┴────────────────────────┘
```